### ISSUE TITLE
KeyError when finding group name

### ISSUE THREAD

**Issue by devosb** [TiroTypeworks/TiroTools#6]
I uncounted a traceback when trying to convert a VOLT file
```
Traceback (most recent call last):
  File "/home/bobby/.local/bin/volto", line 11, in <module>
    load_entry_point('Volto', 'console_scripts', 'volto')()
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 590, in main
    fea = converter.convert()
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 150, in convert
    self._collectStatements(doc)
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 78, in _collectStatements
    self._groupDefinition(statement)
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 197, in _groupDefinition
    glyphs = self._enum(group.enum)
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 184, in _enum
    return ast.GlyphClass(self._coverage(enum.enum))
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 174, in _coverage
    items.append(self._groupName(item))
  File "/srv/bobby/builds/volto3/Lib/volto/__init__.py", line 166, in _groupName
    return ast.GlyphClassName(self._glyphclasses[name.lower()])
KeyError: 'cons'
```
The files used are at https://github.com/silnrsi/font-namdhinggo, The command line (running from the `source` folder) was `volto archive/NamdhinggoSIL1006.vtp master-volto.fea`. I did not use pip3 to install Volto, instead I used `python3 setup.py develop --user`. I can reproduce (exactly) the generated `.fea` from both `Nutso.vtp` and `Nutso.ttf` so I would suspect that my installation is functional.

**Comment by khaledhosny** [comment]
The group `AllCons` references other groups that are defined after it. I guess this is valid in VOLT but Volto does not handle this well. A simple workaround is to move the definition of `AllCons` after the definition of the groups it references.

**Comment by khaledhosny** [comment]
It is not valid for feature files to reference yet to be defined groups, so fixing this will require changing the order of glyph classes when writing them.

**Comment by tiroj** [comment]
VOLT stores groups in alphabetical order, so reordering as a workaround might require renaming the groups in the VOLT project.

Khaled, how difficult would it be to parse the group contents for this kind of reference, and reorder in the .fea code conversion?

**Comment by bobh0303** [comment]
What I did in my [similar] tool was to sort the groups alphabetically (so the FEA was more friendly), then before outputing a group I verify I've already output all the groups it contains. If not I set it aside for later and go on to the next group. Eventually (strategies vary but I took a simple one) I re-visit the ones I set aside to see if I can output them yet. Repeat until done.

**Comment by khaledhosny** [comment]
Yes, something like this should work. I’ll look into it once I have some time, unless someone else beat me to it.

**Comment by khaledhosny** [comment]
(I was thinking of some sort function that would sort a group that references other group after them.)

**Comment by madig** [comment]
We have a VOLT source file that contains this snippet:

```
DEF_LOOKUP "isol_arab.ARA" PROCESS_BASE PROCESS_MARKS "NONE" DIRECTION RTL
IN_CONTEXT
 LEFT GROUP "PreHehContext"
END_CONTEXT
AS_SUBSTITUTION
SUB GLYPH "aHeh.alt"
WITH GLYPH "aHeh"
END_SUB
END_SUBSTITUTION
```

Volto blows up on the same line. I personally don't know VOLT, but https://scripts.sil.org/cms/scripts/page.php?site_id=nrsi&id=VOLT_Tutorial sound like `NONE` and `ALL` are special values or something?

**Comment by tiroj** [comment]
Yes, ALL and NONE are special values for the PROCESS_MARKS flag. Not sure why Volto is failing on this lookup. Hopefully @khaledhosny has some insight.

**Comment by khaledhosny** [comment]
The `NONE`/`ALL` issue is a voltLib bug, I applied a workaround here and submitted https://github.com/fonttools/fonttools/pull/1732 to fix the root issue.

**Comment by anthrotype** [comment]
https://github.com/fonttools/fonttools/releases/tag/4.0.2

**Comment by khaledhosny** [comment]
I released Volto 0.1.2 with the fix. The original issue reported here still unfixed though, will look into it later.
